xbmc/filesystem/MusicDatabaseDirectory/MusicFileItemListModifier.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+*      Copyright (C) 2016 Team Kodi
+*      http://xbmc.org
+*
+*  This Program is free software; you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation; either version 2, or (at your option)
+*  any later version.
+*
+*  This Program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with Kodi; see the file COPYING.  If not, see
+*  <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include "MusicFileItemListModifier.h"
+#include "xbmc/filesystem/MusicDatabaseDirectory/DirectoryNode.h"
+#include "FileItem.h"
+#include "xbmc/music/MusicDbUrl.h"
+#include "xbmc/settings/Settings.h"
+#include "xbmc/guilib/LocalizeStrings.h"
+#include "settings/AdvancedSettings.h"
+
+using namespace XFILE::MUSICDATABASEDIRECTORY;
+
+bool CMusicFileItemListModifier::CanModify(const CFileItemList &items) const
+{
+  return true;
```

Comment:
You're right. This should probably be adjusted in `CFileItemListModification::Modify()` to first check if an `IFileItemListModifier` can actually modify a list of items and only call `Modify()` if that's the case. But this can be done separately.